pkg/agent/validations.go

```diff
@@ -63,6 +62,129 @@ type validationTrace struct {
 
 var previousValidations []validationTrace
 
+type validationResultHistory struct {
+	numFailures     map[string]int
+	seen            map[string]bool
+	currentStatus   map[string]string
+	currentMessage  map[string]string
+	previousStatus  map[string]string
+	previousMessage map[string]string
+}
+
+func checkValidations(cluster *models.Cluster, validationHistory map[string]*validationResultHistory, log *logrus.Logger) error {
+	if validationHistory["cluster"] == nil {
+		validationHistory["cluster"] = &validationResultHistory{
+			numFailures:     make(map[string]int),
+			seen:            make(map[string]bool),
+			currentStatus:   make(map[string]string),
+			currentMessage:  make(map[string]string),
+			previousStatus:  make(map[string]string),
+			previousMessage: make(map[string]string),
+		}
+	}
+	clusterLogPrefix := "Cluster validation: "
+	clusterValidationHistory, err := updateValidationsResultHistory(clusterLogPrefix, cluster.ValidationsInfo, validationHistory["cluster"], log)
```

Comment:
As an opportunity to simplify the code, the function signature could be modified to 

`func updateValidationsResultHistory(logPrefix string, validationsInfoString string, historyMap map[string]*validationResultHistory, key string, log *logrus.Logger) error {`

so that the content of the map could  be update directly within the function (and the element created if it doesn't exist)